test/BUILD

```diff
@@ -273,6 +273,21 @@ plz_e2e_test(
     expected_failure = True,
 )
 
+# Test add_data functionality
+genrule(
+    name = "_add_data_gen",
+    cmd = "echo hello > _add_data_gen.txt",
+    pre_build = lambda name: add_data(name, "add_data_gen.txt"),
```

Comment:
This isn't going to work unless you're using a Please version that has `add_data()`. As you can see, the tests pass on CI because they build you a new version of Please before running this test. 

You can `plz plz test` this, however it might be better to use a `please_repo_e2e_test()`. These can run the development version of Please in a test repo so you don't have to worry about this sort of stuff. There are plenty of examples but five me a shout if you want a hand.